doc/company.texi

```diff
@@ -0,0 +1,1499 @@
+backends act as one complex backend.  @xref{Grouped Backends}.}.  The
+backends are invoked one by one, in the sequential order of the items
+on the @code{company-backends} list.
+
+@cindex active backend
+@cindex backend
+@findex company-diag
+The name of the currently active backend is shown in the mode line and
+in the output of the command @kbd{M-x company-diag}.
+
+@cindex next backend
+@cindex backend
+@findex company-other-backend
+In most cases (mainly to exclude false-positive results), the next
+backend is not invoked automatically.  For the purpose of invoking the
+next backend, use the command @w{@kbd{company-other-backend}}: either
+by calling it with @kbd{M-x} or by binding the command to the keys of
+your choice, such as:
+
+@lisp
+(global-set-key (kbd "C-c C-/") #'company-other-backend)
+@end lisp
+
+@findex company-begin-backend
+It is also possible to specifically start a backend with the command
+@w{@kbd{M-x company-begin-backend}} or by calling a backend by its
+name, for instance: @w{@kbd{M-x company-capf}}.  As usual for Emacs,
+such backends calls can be assigned to key bindings, for example:
+
+@lisp
+(global-set-key (kbd "C-c y") 'company-yasnippet)
+@end lisp
+
+@node Grouped Backends
+@section Grouped Backends
+
+@cindex grouped backends
+@cindex backends
+@vindex company-backends
+In many cases, it can be desirable to receive candidates from several
+backends simultaneously.  This can be achieved by configuring
+@dfn{grouped backends}: a sub-list of backends in the
+@code{company-backends} list, that is handled specifically by Company.
+
+The most important part of this handling is the merge of the
+completion candidates from the grouped backends.  (But only from the
+backends that return the same @emph{prefix} value, see @code{C-h v
+company-backends} for more details.)
+
+To keep the candidates organized in accordance with the grouped
+backends order, add the keyword @code{:separate} to the list of the
+grouped backends.  The following example illustrates this.
+
+@lisp
+(defun my-text-mode-hook ()
+  (setq-local company-backends
+              '((company-dabbrev company-ispell :separate)
+                company-files)))
+
+(add-hook 'text-mode-hook #'my-text-mode-hook)
+@end lisp
+
+Another keyword @code{:with} helps to make sure the results from
+major/minor mode agnostic backends (such as @emph{company-yasnippet},
+@emph{company-dabbrev-code}) are returned without preventing results
+from context-aware backends (such as @emph{company-capf} or
+@emph{company-clang}).  For this feature to work, put backends
+dependent on a mode at the beginning of the grouped backends list,
+then put a keyword @code{:with}, and only then put context agnostic
+backend(s), as shown in the following concise example:
+
+@lisp
+(setq company-backends '((company-capf :with company-yasnippet)))
+@end lisp
+
+@node Package Backends
+@section Package Backends
+
+@c TODO: Maybe extend some of the backends description
```

Comment:
Maybe.

Reply:
Yes, I want to try extending/changing it a bit. At least to extend the tiniest "company-files" section, so that it wouldn't look that miserable.
The initial idea was to devote a separate subsection to a backend, but that didn't work nicely. The current version of this section was the third - and the best IMO so far - attempt to make it fit in more or less.